.github/workflows/changed_file_trigger_examples_check.yml

```diff
@@ -0,0 +1,128 @@
+name: Test Example
+on:
+  pull_request:
+  schedule:
+    # run at 00:00 of every Sunday(singapore time) so here is UTC time Saturday 16:00
+    - cron:  '0 16 * * 6'
+
+jobs:
+# This is for changed files detect and check.
+  detect-changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      changedFiles: ${{ steps.changed-files.outputs.all_changed_files }}
+    name: Check out all files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Get all changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v35
+        with:
+          since_last_remote_commit: true  # Using this can trigger action each time a PR is submitted.
+
+  prepare-changed-file-matrix:
+    name: Prepare changed file List for checking
+    needs: detect-changed-files
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: setup matrix
+      id: set-matrix
+      run: |
+          changedFileName=""
+          for file in ${{ needs.detect-changed-files.outputs.changedFiles  }}; do
+            changedFileName="${file}:${changedFileName}"
+          done
+          echo "$changedFileName was changed"
+          res=`python .github/workflows/changed_file.py --fileNameList $changedFileName`
+          echo "All changed files are $res"
+          if [ -z "$res" ]
+          then
+                echo "\No change in example folder!"
+          else
+                var=$( IFS=',' ; echo "${res[*]}" )
+                echo "$var"
+                echo "::set-output name=matrix::{\"var\":$(echo "$var")}"
+          fi
+
+# If no file is changed, it will prompt an error and shows the matrix do not have value.
+  check-all-changed-files:
+    if:
+    name: Test each changed example files
+    needs: prepare-changed-file-matrix
+    runs-on: [self-hosted, gpu]
+    strategy:
+      matrix: ${{fromJson(needs.prepare-changed-file-matrix.outputs.matrix)}}
```

Comment:
If it runs with the `schedule` event, there is no changed files and there is no need to check for changed files.